code/modules/shuttle/custom_shuttle.dm

```diff
@@ -1,124 +1,132 @@
-	var/deltaX = port.x - x
-	var/deltaY = port.y - y
-	var/deltaZ = (port.z - z) * Z_DIST
-	return sqrt(deltaX * deltaX + deltaY * deltaY + deltaZ * deltaZ) * distance_multiplier
+/obj/machinery/computer/shuttle_flight/custom_shuttle/ui_static_data(mob/user)
+	var/list/data = ..()
+	calculateStats()
+	data["display_fuel"] = TRUE
+	return data
+
+/obj/machinery/computer/shuttle_flight/custom_shuttle/ui_data(mob/user)
+	var/list/data = ..()
+	data["fuel"] = get_fuel()
+	data["display_stats"] = list(
+		"Shuttle Mass" = "[calculated_mass/10] Tons",
+		"Engine Force" = "[calculated_dforce]kN ([calculated_engine_count] engines)",
+		"Supercruise Acceleration" = "[calculated_acceleration] ms^-2",
+		"Fuel Consumption" = "[calculated_consumption] moles per second",
+		"Engine Cooldown" = "[calculated_cooldown] seconds"
+	)
+	if(calculated_acceleration < 1)
+		data["thrust_alert"] = "Insufficient engine power at last callibration. Launch shuttle to recalculate thrust."
+	else
+		data["thrust_alert"] = 0
+	if(calculated_non_operational_thrusters > 0)
+		data["damage_alert"] = "[calculated_non_operational_thrusters] thrusters offline."
+	else
+		data["thrust_alert"] = 0
+	return data
+
+/obj/machinery/computer/shuttle_flight/custom_shuttle/launch_shuttle()
+	calculateStats()
+	if(calculated_acceleration < 1)
+		say("Insufficient engine power to engage supercruise.")
+		return
+	var/datum/orbital_object/shuttle/custom_shuttle/shuttle = ..()
+	if(shuttle)
+		shuttle.attached_console = src
+	return shuttle
+
+//Consumes fuel and reduces thrust of engines that run out of fuel
+/obj/machinery/computer/shuttle_flight/custom_shuttle/proc/consume_fuel(var/multiplier = 1)
+	//Reset stats
+	calculated_dforce = 0
+	calculated_acceleration = 0
+	calculated_engine_count = 0
+	calculated_consumption = 0
+	//Consume fuel
+	for(var/obj/machinery/shuttle/engine/E as() in shuttle_engines)
+		//Check for inop engines
+		if(!E.attached_heater)
+			continue
+		var/obj/machinery/atmospherics/components/unary/shuttle/heater/shuttle_heater = E.attached_heater.resolve()
+		if(!shuttle_heater)
+			continue
+		if(shuttle_heater.dir != E.dir)
+			continue
+		if(shuttle_heater.panel_open)
+			continue
+		if(!shuttle_heater.anchored)
+			continue
+		//Setup correct, check fuel.
+		if(shuttle_heater.hasFuel(E.fuel_use * multiplier))
+			shuttle_heater.consumeFuel(E.fuel_use * multiplier)
+			calculated_consumption += E.fuel_use
+			calculated_dforce += E.thrust
+			calculated_engine_count ++
+	calculated_acceleration = (calculated_dforce*1000) / (calculated_mass*100)
+
+/obj/machinery/computer/shuttle_flight/custom_shuttle/proc/check_stranded()
+	if(!calculated_engine_count && shuttleObject)
+		say("Fuel reserves depleted, dropping out of supercruise.")
+		if(!shuttleObject.docking_target)
+			if(shuttleObject.can_dock_with)
+				shuttleObject.docking_target = shuttleObject.can_dock_with
+			else
+				//Send shuttle object to random location
```

Comment:
A bold choice...

Reply:
It only sends them to beacons, which are random empty / space ruin z-levels, unless the shuttle is above a docking location in which case its sent to that z-level.